sdk/xmake.lua

```diff
@@ -253,21 +253,143 @@ target("cheriot.software_revoker")
 		target:set("cheriot.ldscript", "software_revoker.ldscript")
 	end)
 
--- Helper to get the board file for a given target
-local board_file = function(target)
-	local boardfile = target:values("board")
-	if not boardfile then
-		raise("target " .. target:name() .. " does not define a board name")
-	end
+-- Helper to find a board file given either the name of a board file or a path.
+local function board_file_for_name(boardName)
+	local boardfile = boardName
 	-- The directory containing the board file.
 	local boarddir = path.directory(boardfile);
+	-- If this isn't a path, look in the boards directory
 	if path.basename(boardfile) == boardfile then
 		boarddir = path.join(scriptdir, "boards")
-		boardfile = path.join(boarddir, boardfile .. '.json')
+		local fullBoardPath = path.join(boarddir, boardfile .. '.json')
+		if not os.isfile(fullBoardPath) then
+			fullBoardPath = path.join(boarddir, boardfile .. '.patch')
+		end
+		if not os.isfile(fullBoardPath) then
+			print("unable to find board file " .. boardfile .. ".  Try specifying a full path")
+			return nil
+		end
+		boardfile = fullBoardPath
 	end
 	return boarddir, boardfile
 end
 
+-- Helper to get the board file for a given target
+local function board_file_for_target(target)
+	local boardName = target:values("board")
+	if not boardName then
+		print("target " .. target:name() .. " does not define a board name")
+		return nil
+	end
+	return board_file_for_name(boardName)
+end
+
+-- Helper to load a board file.  This must be passed the json object provided
+-- by import("core.base.json") because import does not work in helper
+-- functions at the top level.
+local function load_board_file(json, boardFile)
+	if path.extension(boardFile) == ".json" then
+		return json.loadfile(boardFile)
+	end
+	if path.extension(boardFile) ~= ".patch" then
+		print("unknown extension for board file: " .. boardFile)
+		return nil
+	end
+	local patch = json.loadfile(boardFile)
+	if not patch.base then
+		print("Board file " .. boardFile .. " does not specify a base")
+		return nil
+	end
+	local _, baseFile = board_file_for_name(patch.base)
+	local base = load_board_file(json, baseFile)
+
+	-- If a string value is a number, return it as number, otherwise return it
+	-- in its original form.
+	function asNumberIfNumber(value)
```

Comment:
Hrm.  This function isn't just `tonumber(value) or value` (numeric 0 is _true_ in Lua, and `tonumber` won't return `false`) because we want to keep things Lua can't exactly represent as strings?  I don't think we're likely to _notice_ the difference in practice, but defense in depth is probably a good idea.

Reply:
I’m not sure I understand. JSON keys are strings for objects or numbers for dictionaries. Will this do the wrong thing if a key is called “true”?

Reply:
Lua's a strange beast.  `tonumber("true") or "true"` is the string `"true"`, not the boolean `true`, because `nil or X` evaluates to `X` unchanged, and `X or Y` with `X` neither `nil` or `false` evaluates to `X`.

Reply:
That is the desired behaviour. Booleans do not want this kind of special casing, only numbers (technically, only integers). But given the ability to accurately tell if a JSON thing is an array, I don’t think this function is needed.